### PR TITLE
Added annotation typing to shufflenet

### PR DESCRIPTION
**Pull request by frgfm**
Hi there!

As per #2025, annotation typing are welcome in torchvision. So, this PR focuses on `torchvision.models.shufflenet` this PR!

Any feedback is welcome!

### CODE REVIEW

File: torchvision/models/shufflenetv2.py [hunk 1]
```diff
@@ -68,10 +74,11 @@ def __init__(self, inp, oup, stride):
         )
 
     @staticmethod
-    def depthwise_conv(i, o, kernel_size, stride=1, padding=0, bias=False):
+    def depthwise_conv(i: int, o: int, kernel_size: int, stride: int = 1, padding: int = 0,
+                       bias: bool = False) -> nn.Conv2d:
```
Comment (pmeier) [inline torchvision/models/shufflenetv2.py:78]: Could you format this with `black` to make it more legible?

**Review by pmeier — COMMENTED** [review]
Minor nitpick plus incremental changes from the other PRs. Otherwise LGTM.

File: torchvision/models/shufflenetv2.py (hunk 1 above)
Comment (frgfm) [reply torchvision/models/shufflenetv2.py]: Fixed!

**Comment by frgfm** [comment]
Still getting a troublesome error with mypy though:
```
torchvision/models/shufflenetv2.py:148: error: "Tensor" not callable  [operator]
            x = self.stage2(x)
                ^
torchvision/models/shufflenetv2.py:149: error: "Tensor" not callable  [operator]
            x = self.stage3(x)
                ^
torchvision/models/shufflenetv2.py:150: error: "Tensor" not callable  [operator]
            x = self.stage4(x)
```
Any ideas? I tried a few things in vain unfortunately

**Comment by fmassa** [comment]
I'll let @pmeier have a look, I don't know why mypy is thinking that `stage2` is a Tensor. One thing to note is that `stage2` is created programmatically with `setattr`, which might confuse mypy. Maybe there is a way to annotate `stage{n}` to tell mypy that it's a `nn.Module` ?

**Comment by pmeier** [comment]
>  I don't know why mypy is thinking that stage2 is a Tensor

Me neither. I think that might be caused by the way `nn.Module` is typed, but I'm certainly not sure.

> One thing to note is that `stage2` is created programmatically with `setattr`, which might confuse mypy. Maybe there is a way to annotate `stage{n}` to tell mypy that it's a `nn.Module` ?

We could simply include `self.stage2: nn.Module` in the constructor. Since `mypy` only inspects but does **not** the code we can't use a loop for that. Thus I propose to wrap the loop in a function an call it several times from the constructor.

**Comment by fmassa** [comment]
> Thus I propose to wrap the loop in a function an call it several times from the constructor

I'm not exactly sure how to do it, given that we use `setattr`. So you propose to make the change that we instead do by hand
```python
self.stage2 = create(...)
```
?

That is a big big change for just typing, can't we disable mypy somehow?

**Comment by pmeier** [comment]
> can't we disable mypy somehow?

Sure we can. Maybe `cast`ing would be easier:

```python
x = cast(nn.Sequential, self.stage2)(x)
```

**Review by fmassa — APPROVED** [review]
Thanks!

File: torchvision/models/shufflenetv2.py [hunk 2]
```diff
@@ -104,6 +123,10 @@ def __init__(self, stages_repeats, stages_out_channels, num_classes=1000, invert
 
         self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)
 
+        # Static annotations for mypy
+        self.stage2: nn.Sequential
+        self.stage3: nn.Sequential
+        self.stage4: nn.Sequential
```
Comment (fmassa) [inline torchvision/models/shufflenetv2.py:129]: note: it would have been preferable to let `stage{}` be a `nn.Module`, as they could potentially be any type of `nn.Module`, not just `Sequential`
